api/desecapi/serializers.py

```diff
@@ -143,6 +153,69 @@ def update(self, instance: RRset, validated_data):
         return instance
 
 
+class RRsetListSerializer(ListSerializer):
+    def create(self, validated_data):
+        index = {(data.get('subname', ''), data['type']) for data in validated_data}
+        for (subname, type_) in index:
+            if not index[(subname, type_)].get('type', None):
+                raise ValidationError('type is required')
+            if not index[(subname, type_)].get('ttl', None):
```

Comment:
Better: check `is not None`. If what I wrote above is right, then the current check also triggers when ttl is 0, but it seems like you really want to see if `get('ttl')` was successful.